robot/impl/resource_manager.go

```diff
@@ -778,7 +778,7 @@ func (manager *resourceManager) completeConfig(
 func (manager *resourceManager) completeConfigForRemotes(ctx context.Context, lr *localRobot) {
 	for _, resName := range manager.resources.FindNodesByAPI(client.RemoteAPI) {
 		gNode, ok := manager.resources.Node(resName)
-		if !ok || !gNode.NeedsReconfigure() {
+		if !ok || !gNode.NeedsReconfigure() || gNode.MarkedForRemoval() {
```

Comment:
can you double check if this check is necessary with the other changes? if we correctly stop allowing nodes to transition to unhealthy, then this check should be unnecessary - there should not be a way for nodes to be both needs reconfigure and marked for removal at the same time

Reply:
yeah we can totally remove this. Sorry was working on fixing tests with the new graph node changes. Was gonna do another look over the pr before retagging you